File: projects/Янчук/4/LABB4NEW4/app/src/main/java/com/example/labb4new4/CartActivity.java
package com.example.labb4new4;

import android.os.Bundle;
import android.widget.ListView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;

public class CartActivity extends AppCompatActivity {

    ArrayList<Product> cart;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_cart);

        cart = (ArrayList<Product>) getIntent().getSerializableExtra("cart");

        ListView listView = findViewById(R.id.cartList);
        listView.setAdapter(new CartAdapter(this, cart));
    }
}

File: projects/Гончеренок/lab4/src/MiniShop/app/src/main/java/com/example/minishop/CartActivity.kt
package com.example.minishop

import android.os.Build
import android.os.Bundle
import android.widget.ListView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class CartActivity : AppCompatActivity() {

    private lateinit var listView: ListView
    private lateinit var tvCartTitle: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_cart)

        listView = findViewById(R.id.listViewCart)
        tvCartTitle = findViewById(R.id.tvCartTitle)

        val items = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            intent.getSerializableExtra("checked_items", ArrayList::class.java) as? ArrayList<Product>
        } else {
            @Suppress("DEPRECATION")
            intent.getSerializableExtra("checked_items") as? ArrayList<Product>
        } ?: arrayListOf()

        tvCartTitle.text = "Корзина товаров (${items.size})"
        listView.adapter = CartAdapter(this, items)
        title =""
    }
}